lib/elixir/lib/kernel.ex

```diff
@@ -795,11 +798,13 @@ defmodule Kernel do
 
       iex> trunc(5.4)
       5
+
       iex> trunc(5.99)
       5
 
   """
-  @spec trunc(number) :: integer
+  @spec trunc(arg) :: arg when arg: integer
```

Comment:
I think when they say "returns the same tuple" they really mean "returns tuple of the same type". So if you're given `{integer(), float()}` you are going to return the same type but values of the integer and float can be anything.

If they really mean the same exact value a lot of typespecs would not make sense. For example take a look at `lists:map/2` http://erlang.org/doc/man/lists.html#map-2. How would that make sense if all instances of `A` and `B` were the same value?